src/ucp/api/ucp.h

```diff
@@ -1425,29 +1443,30 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
 /**
  * @ingroup UCP_ENDPOINT
  *
- * @brief Initiate non-blocking disconnect.
+ * @brief Non-blocking @ref ucp_ep_h "endpoint" closure.
  *
- *   This routine starts a disconnect process which would eventually release the
- * @ref ucp_ep_h "endpoint". The disconnect process flushes, locally, all
- * outstanding communications, and releases all memory contexts associated with
- * the endpoint. After calling this function, the endpoint cannot be used anymore.
- *   Nevertheless, if the application is interested to re-initiate communication
- * with a particular remote worker, it can use @ref ucp_ep_create "endpoints
- * create routine" to re-open a new endpoint.
+ * This routine releases the @ref ucp_ep_h "endpoint". The endpoint closure
+ * process depends on set @a flags.
  *
- * @param [in]  ep   Handle to the endpoint to disconnect.
+ * @param [in]  ep      Handle to the endpoint to close.
+ * @param [in]  mode    One from @ref ucp_ep_close_nb_mode "enumerator" value.
  *
- * @return UCS_OK           - The endpoint is flushed and destroyed.
- * @return UCS_PTR_IS_ERR(_ptr) - The disconnect operation failed.
- * @return otherwise        - The disconnect process started, and can be
+ * @return UCS_OK           - The endpoint is closed successfully.
+ * @return UCS_PTR_IS_ERR(_ptr) - The closure failed, error code indicates
+ *                                transport level status. Resources are released,
+ *                                so the endpoint can't be used.
```

Comment:
Suggested rewording:
The closure failed and an error code indicates the transport level status. However, resources
are released and the endpoint can no longer be used.